packages/api/src/external/hie/validate-patient-links.ts

```diff
@@ -38,38 +41,67 @@ export async function validateCwLinksBelongToPatient(
   cwLinks: CwLink[],
   patientData: PatientData
 ): Promise<{ validNetworkLinks: CwLink[]; invalidLinks: CwLink[] }> {
-  const validNetworkLinks: CwLink[] = [];
-  const invalidLinks: CwLink[] = [];
+  const linkToPatientDataMap = cwLinks.map(cwLink => ({
+    link: cwLink,
+    patientData: cwLinkToPatientData(cwLink),
+  }));
 
-  for (const cwLink of cwLinks) {
-    const linkPatientData = cwLinkToPatientData(cwLink);
+  const { validPatientDataLinks, invalidPatientDataLinks } = await validateLinksBelongToPatient(
+    cxId,
+    patientData,
+    linkToPatientDataMap.map(item => item.patientData)
+  );
 
-    const isPatientMatch = await validateLinkBelongsToPatient(cxId, linkPatientData, patientData);
+  const validNetworkLinks = linkToPatientDataMap
+    .filter(item => validPatientDataLinks.includes(item.patientData))
+    .map(item => item.link);
 
-    if (isPatientMatch) {
-      validNetworkLinks.push(cwLink);
-    } else {
-      invalidLinks.push(cwLink);
-    }
-  }
+  const invalidLinks = linkToPatientDataMap
+    .filter(item => invalidPatientDataLinks.includes(item.patientData))
+    .map(item => item.link);
 
   return { validNetworkLinks, invalidLinks };
 }
 
-const validateLinkBelongsToPatient = async (
+async function validateLinksBelongToPatient(
```

Comment:
these are mutually exclusive and the parent ultimately is deciding what is actually a valid link / invalid link.

Can we return only invalid links here and then use partition in the caller to minimize the diff variables we have floating around?

```
  const [invalidItems, validItems] = partition(
    linkToPatientDataMap,
    item => invalidPatientDataLinks.includes(item.patientData)
  );
```